src/python/WMCore/ReqMgr/CherryPyThreads/StatusChangeTasks.py

```diff
@@ -6,9 +6,76 @@
 
 from Utils.CherryPyPeriodicTask import CherryPyPeriodicTask
 from WMCore.Services.WMStats.WMStatsReader import WMStatsReader
+from WMCore.Services.WorkQueue.WorkQueue import WorkQueue
 from WMCore.Services.RequestDB.RequestDBWriter import RequestDBWriter
 from WMCore.Services.WMStats.DataStruct.RequestInfoCollection import RequestInfoCollection
 
+
+def moveForwardStatus(reqDBWriter, requestStatus):
+    
+    statusTransition = {"assigned": ["acquired", "running-open", "running-closed", "completed"], 
```

Comment:
Yes, it is not the sam state transition path (In this case there is one path which WorkQueue can control).
You are right about moving this to RequestStatus file. I will do that.